src/posthog-surveys.ts

```diff
@@ -214,34 +221,50 @@ export class PostHogSurveys {
             return
         }
 
-        const generateSurveys = phExtensions.generateSurveys
-
         if (!this._decideServerResponse) {
             logger.warn('Decide not loaded yet. Not loading surveys.')
             return
         }
 
-        if (this._surveyEventReceiver == null) {
-            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
-        }
-
-        if (!generateSurveys) {
-            const loadExternalDependency = phExtensions.loadExternalDependency
-
-            if (loadExternalDependency) {
-                loadExternalDependency(this.instance, 'surveys', (err) => {
-                    if (err) {
-                        logger.error('Could not load surveys script', err)
-                        return
-                    }
-
-                    this._surveyManager = phExtensions.generateSurveys?.(this.instance)
-                })
+        this._isInitializingSurveys = true
+
+        try {
+            const generateSurveys = phExtensions.generateSurveys
+
+            if (!generateSurveys) {
+                const loadExternalDependency = phExtensions.loadExternalDependency
+
+                if (loadExternalDependency) {
+                    loadExternalDependency(this.instance, 'surveys', (err) => {
+                        if (err) {
+                            logger.error('Could not load surveys script', err)
+                            this._isInitializingSurveys = false
+                            return
+                        }
+
+                        this._surveyManager = phExtensions.generateSurveys?.(this.instance)
+                        this._isInitializingSurveys = false
+                        if (this._surveyEventReceiver == null) {
+                            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
+                        }
+                        logger.info('Surveys loaded successfully')
+                    })
+                } else {
+                    logger.error('PostHog loadExternalDependency extension not found. Cannot load remote config.')
+                    this._isInitializingSurveys = false
+                }
             } else {
-                logger.error('PostHog loadExternalDependency extension not found. Cannot load remote config.')
+                this._surveyManager = generateSurveys(this.instance)
```

Comment:
same as https://github.com/PostHog/posthog-js/pull/1747/files#r1964979053

Reply:
for this one, it actually guarantees that it's not undefined. since we first check if `generateSurveys` is undefined on [this if](https://github.com/PostHog/posthog-js/pull/1747/files#diff-e2c6b58ed91904007d3de10b7836a7dc30e4239c8e4d88ccd9f705e061920e88R232-R234). so if it is, we guarantee that it has a value